components/prism-graphql.js

```diff
@@ -27,17 +37,44 @@ Prism.languages.graphql = {
 		pattern: /[a-z_]\w*(?=\s*(?:\((?:[^()"]|"(?:\\.|[^\\"\r\n])*")*\))?:)/i,
 		greedy: true
 	},
+	'constant': /\b(?!ID\b)[A-Z][A-Z_\d]*\b/,
 	'class-name': {
-		pattern: /(\b(?:enum|implements|interface|on|scalar|type|union)\s+|&\s*)[a-zA-Z_]\w*/,
-		lookbehind: true
+		pattern: /(\b(?:enum|implements|interface|on|scalar|type|union)\s+|&\s*|:\s*|\s*\[)[A-Z_]\w*/,
+		lookbehind: true,
+		greedy: true
 	},
 	'fragment': {
 		pattern: /(\bfragment\s+|\.{3}\s*(?!on\b))[a-zA-Z_]\w*/,
 		lookbehind: true,
 		alias: 'function'
 	},
-	'keyword': /\b(?:directive|enum|extend|fragment|implements|input|interface|mutation|on|query|repeatable|scalar|schema|subscription|type|union)\b/,
+	'property-mutation': {
+		pattern: /\b(mutation.*(?:[\n\r\u2028\u2029][\n\s]*)?)\b\w+/g,
+		lookbehind: true,
+		greedy: true
+	},
+	'definition-mutation': {
+		pattern: /(\bmutation\s+|\.{3}\s*)[a-zA-Z_]\w*/,
+		lookbehind: true,
+		alias: 'function'
+	},
+	'definition-query': {
+		pattern: /(\bquery\s+|\.{3}\s*)[a-zA-Z_]\w*/,
+		lookbehind: true,
+		alias: 'function'
+	},
+	'keyword-mutation': {
+		pattern: /\b(?:mutation)\b/,
+		alias: 'keyword'
+	},
+	'keyword-query': {
+		pattern: /\b(?:query)\b/,
+		alias: 'keyword'
+	},
```

Comment:
I don't think this is necessary. We have [a plugin](https://prismjs.com/plugins/highlight-keywords/) just for this purpose.

Reply:
I see, that's pretty cool. How do I say that the `graphql` language requires this plugin? Or is that up to the end user?

I ask because I would like to keep the integration tests I have passing, but I can concede it's reasonable that the plugin is separately tested so if someone just imports that plugin they can count on it working.

Reply:
> Or is that up to the end user?

It's supposed to be on the end user.

Technically you _could_ require the plugin but please don't. The plugin will add these keyword aliases for each and every language. GraphQL would essentially force this behavior onto other languages.

> if someone just imports that plugin they can count on it working.

Yup, that's the idea. The user just has to load the file and it'll work.

> I would like to keep the integration tests I have passing

With "integration tests" you mean the `.test` files?

If so, sorry but that's won't work because the test files contain the token names. It's really easy to update test files tho (`npm run test:languages -- --update`), so this shouldn't be a problem.

Reply:
> The plugin will add these keyword aliases for each and every language. 

Right, I would love to isolate it to graphql if possible:

```
	Prism.hooks.add('wrap', function wrapGraphqlKeywords(env) {
		if (env.type === 'keyword' && env.language === 'graphql') {
			env.classes.push('keyword-' + env.content);
		}
	});
```

but it didn't work when I put it in the `components/prism-graphql.js` file, even in an IIFE. I'm missing some understanding about how plugins work in the system, could you explain that to me? :)

Reply:
Or, is it just not worth it? Probably that eh? 😅

Reply:
> Just to confirm, to add the plugin, I could just require it like:

In NodeJS? Yes.

> but it didn't work

Oh, it does work. `env.classes` is essentially the list of aliases, so you are adding a special alias for each keyword. However, our test file format doesn't include aliases, so they won't show up.

That is one reason why it doesn't work. The other is that the `wrap` hook is called after tokenization. So any and all changes a `wrap` hook does do not affect our test files which only store tokens.

You have to update the tests either way...

Reply:
Ah. Thanks for the explanation!

Reply:
I came up with a quick and dirty way to write some "e2e tests", would love to know what you think :) https://github.com/JaKXz/prism/pull/1/files

Reply:
^with those tests in I've resolved this 👍🏽